execution/executor-types/src/executed_chunk.rs

```diff
@@ -114,6 +114,18 @@ impl ExecutedChunk {
         }
     }
 
+    pub fn ensure_ends_with_state_checkpoint(&self) -> Result<()> {
+        ensure!(
+            self.next_epoch_state.is_some()
```

Comment:
why is it?

Reply:
Just being defensive.. we assume all blocks end with state checkpoints for now.